src/backend/renderer/sync/mod.rs

```diff
@@ -0,0 +1,90 @@
+//! Helper for synchronizing rendering operations
+use std::{os::unix::io::OwnedFd, sync::Arc};
+
+use downcast_rs::{impl_downcast, Downcast};
+
+#[cfg(feature = "backend_egl")]
+mod egl;
+
+/// A fence that will be signaled in finite time
+pub trait Fence: std::fmt::Debug + Send + Sync + Downcast {
+    /// Queries the state of the fence
+    fn is_signaled(&self) -> bool;
```

Comment:
I wonder if we could avoid needing to clone a bunch of state for the vulkan renderer by making checks for signalling be done on the renderer/device. SyncPoint however makes that a little more complicated to do.

Essentially my opinion is that a fence implementation should be a sort of handle into the renderer/device.

Reply:
Thinking about the implementation, I would probably be able to make this work with a `Weak<ash::Device>` and an underlying vulkan `Fence` (with a drop of sort to ensure the underlying fence doesn't go away while a fence live) (pre-exporting ahead of time). So that I don't need to clone a whole lot of the renderer.

This would also mean that it would still be Send + Sync